src/pages/library/Datasets.js

```diff
@@ -427,11 +429,24 @@ const topMed = () => h(Participant, {
   }, ['Browse Data'])
 ])
 
+const DatasetsAuthWrapper = () => {
```

Comment:
it's a bit of a hack to ensure the Datasets component rerenders when a new user login completes

Reply:
Ah, because `getLocalPref('catalog-toggle')` depends on the current user ID and needs to be re-rendered when it changes?

I think this would be worth a comment since it's not obvious.

Also, I suspect `useStore(authStore)` could be placed in the `Datasets` component instead of a wrapper component. And the return value doesn't need to be stored in a variable.

Reply:
Then again, since Datasets is a child of Router, which is a child of AuthContainer and AuthContainer subscribes to changes in the auth store, does this not already re-render when the user changes?

Reply:
Oh! When we were pairing I missed that the Router will trigger a re-render when the user changes. That's great in this case!

We can't put `useStore(authStore)` in `Datasets`. I think we tried that first and it didn't work, which is why we moved to this approach. We wanted a complete re-instantiation of the component, rather than a partial re-rendering.

Reply:
@nawatts looks like you're right here, it turned out to not be necessary, thanks!

Reply:
A change to props or state causes the component to render. React then reconciles the DOM with the new element tree returned by that render.

https://reactjs.org/docs/faq-state.html#what-is-the-difference-between-state-and-props
https://reactjs.org/docs/reconciliation.html